File: BEMLIB_22.10/03_laplace/lgf_2d/lgf_2d_fs.m
 function [G,Gx,Gy] = lgf_2d_fs ...
...
  (x,y ...
  ,x0,y0 ...
  ,Iopt)

%=========================================
% FDLIB, CFDLAB, BEMLIB
%
% Copyright by C. Pozrikidis, 1999
% All rights reserved
%
% This program is to be used only under the
% stipulations of the licensing agreement
%=========================================

%-------------------------------------------
% Free-space Green's function:
%
%  G = -(1/2*pi) * ln(r)
%
% Iopt = 1: compute only the Green's function
%     ne 1: compute the Green's function
%           and the gradient
%-------------------------------------------

%----------
% constants
%----------

pi2 = 2*pi;

%-----------
% initialize
%-----------

G = 0;
Gx = 0;
Gy = 0;

%-----------------
% Green's function
%-----------------

dx = x-x0;
dy = y-y0;
rs = dx*dx+dy*dy;
G  = -0.5*log(rs)/pi2;

%---------
% gradient
%---------

if(Iopt>1)

 den = rs*pi2;
 Gx = - dx/den;
 Gy = - dy/den;

end

%-----
% done
%-----

return

File: BEMLIB_22.10/03_laplace/flow_1d/lgf_2d_fs.f
      subroutine lgf_2d_fs 
     +
     +    (Iopt
     +    ,x,y
     +    ,x0,y0
     +    ,G
     +    ,Gx,Gy
     +    )

c========================================
c FDLIB, BEMLIB
c
c Copyright by C. Pozrikidis, 1999
c All rights reserved
c
c This program is to be used only under the
c stipulations of the licensing agreement
c========================================

c----------------------------------------
c Free-space Green's function of Laplace's
c equation in two dimensions:
c
c  G = -1/(2*pi) * lnr
c
c  Iopt =  1 computes only the Green's function
c       ne 1 computes the Green's function
c                     and its gradient
c----------------------------------------

      Implicit Double Precision (a-h,o-z)

c---
c constants
c---

      pi  = 3.1415 92653 58979 32384 D0
      pi2 = 2.0D0*pi

c--- 
c launch
c--- 

      dx = x-x0
      dy = y-y0

      rs = dx*dx+dy*dy

      G  = - 0.5D0*log(rs)/pi2

      if(Iopt.gt.1) then

       den = rs*pi2
       Gx = - dx/den
       Gy = - dy/den

      end if

c-----
c Done
c-----

      return
      end
